### PR TITLE
[GOBBLIN-1391] Filter Kafka topics in kafka consumer client with No s…

### PR DESCRIPTION
**Pull request by vikrambohra**
…chema in schema registry

Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-1391] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1391

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
- Add schema registry to gobblin kafka consumer client.
- filter kafka topics that have no schema in schema registry.

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
- Tested in linkedin holdem.

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java [hunk 1]
```diff
@@ -54,13 +58,14 @@
...
   protected final int socketTimeoutMillis;
+  protected final KafkaSchemaRegistry schemaRegistry;
 
   public AbstractBaseKafkaConsumerClient(Config config) {
     this.brokers = ConfigUtils.getStringList(config, ConfigurationKeys.KAFKA_BROKERS);
     if (this.brokers.isEmpty()) {
       throw new IllegalArgumentException("Need to specify at least one Kafka broker.");
     }
-
+    this.schemaRegistry = KafkaSchemaRegistry.get(ConfigUtils.configToProperties(config));
```
Comment (sv2000) [inline gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java:68]: What happens when Kafka Schema Registry is not enabled? Shouldn't we check if the config kafka.schema.registry.url is set?

Comment (sv2000) [reply gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java:68]: Also - can we do a lazy instantiation of KafkaSchemaRegistry inside getFilteredTopics() method? We only need the schema registry check for filtering the list of topics.

**Review by sv2000 — CHANGES_REQUESTED** [review]

File: gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java (hunk 1 above)
Comment (vikrambohra) [reply gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java:68]: 1. There is check for schema registry class inside the KafkaSchemaRegistry.get method
  public static <K, S> KafkaSchemaRegistry<K, S> get(Properties props) {
    Preconditions.checkArgument(props.containsKey(KAFKA_SCHEMA_REGISTRY_CLASS),
        "Missing required property " + KAFKA_SCHEMA_REGISTRY_CLASS);
...
}
2. I am assuming we should only initialize it "once" lazily too right?

File: gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java [hunk 2]
```diff
@@ -75,14 +81,30 @@ public AbstractBaseKafkaConsumerClient(Config config) {
 
   @Override
   public List<KafkaTopic> getFilteredTopics(final List<Pattern> blacklist, final List<Pattern> whitelist) {
+    final Optional<KafkaSchemaRegistry> schemaRegistry = (config.hasPath(KafkaSchemaRegistry.KAFKA_SCHEMA_REGISTRY_CLASS) && config.hasPath(KafkaSchemaRegistry.KAFKA_SCHEMA_REGISTRY_URL)) ? Optional.of(KafkaSchemaRegistry.get(ConfigUtils.configToProperties(this.config))) : Optional.absent();
```
Comment (sv2000) [inline gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java:84]: Can we cache the schema registry into a member variable and avoid re-initializing it each time it is needed? Also - you can move the check for schema registry existence into a separate method.

**Review by sv2000 — CHANGES_REQUESTED** [review]
One comment. Looks good otherwise.

File: gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java [hunk 3]
```diff
@@ -78,11 +85,33 @@ public AbstractBaseKafkaConsumerClient(Config config) {
     return Lists.newArrayList(Iterables.filter(getTopics(), new Predicate<KafkaTopic>() {
       @Override
       public boolean apply(@Nonnull KafkaTopic kafkaTopic) {
-        return DatasetFilterUtils.survived(kafkaTopic.getName(), blacklist, whitelist);
+        return DatasetFilterUtils.survived(kafkaTopic.getName(), blacklist, whitelist) && isSchemaPresent(kafkaTopic.getName());
       }
     }));
   }
 
+  private boolean isSchemaRegistryPresent() {
```
Comment (sv2000) [inline gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java:93]: Nit: would recommend renaming the method to isSchemaRegistryConfigured.

**Review by sv2000 — CHANGES_REQUESTED** [review]
A comment to rename a method. But looks good overall.

**Review by sv2000 — APPROVED** [review]
+1.

**Pull request closed without merging**